package ProtoBuilding;

message BuildInfo
{
	required uint32 id 			= 1; //建筑id
	required uint32 level		= 2; //当前等级
	required uint32 status		= 3; //状态[-1>未解锁;0>正常状态;1>解锁中;2>升级中]
	required uint32 handle_time	= 4; //此操作起始时间
	required bool	is_speed	= 5; //是否加速
}

message BuildQueue
{
	required uint32 left_cnt		= 1; //建筑自动升级剩余次数
	required uint32	type			= 2; //建筑自动升级类型
	repeated int32 	queue			= 3; //建筑升级队列[>0升级任务;=0空闲;-1未解锁]
}

message BuildUserSyncInfo
{
	required uint32		total_exploit	= 1; //累计功勋
	required uint32		user_coins		= 2; //当前银币
	required uint32 	user_woods		= 3; //当前木材
	required uint64		user_exp		= 4; //当前玩家经验
	required uint32		user_level		= 5; //当前玩家等级
}

message BuildYield
{
	required uint32 type 			= 1; //建筑type
	required uint32 capacity		= 2; //容量  兵营返回0
	required uint32	yield			= 3; //产量  兵营单位 5s 其他为 1小时
}

message BuildSyncReq
{
	required uint32		type			= 1; //[0>全部; 非0>对应的的建筑类型]
}

message BuildSyncResp
{
	required uint32		type			= 1;//[0>全部;非0>对应的的建筑类型]
	repeated BuildInfo 	builds 			= 2;  //
	required BuildQueue queue		 	= 3;
	required BuildUserSyncInfo user	  	= 4;
}

//建筑解锁
message BuildUnlockReq
{
	repeated uint32 	ids    			= 1; //需要解锁的建设
	optional bool		lock_queue		= 2; //需要解锁的建筑队列[没填写或者false为不解锁]
	
}

//
message BuildUnlockResp
{
	required BuildQueue	build_queue		= 1; //当前建造队列	
	repeated BuildInfo 	builds 			= 2; //建筑[数组]	
}

//解锁建筑队列
message BuildUnlockQueueReq
{
	
}

message BuildUnlockQueueResp
{
	required BuildQueue	build_queue		= 1; //当前建造队列
	repeated BuildInfo 	builds 			= 2; //建筑[数组]	
}



//建筑升级
message BuildUpgradeReq
{
	required uint32 	id    			= 1;
}

message BuildUpgradeResp
{
	required uint32 			id    			= 1; //建筑Id
	required BuildUserSyncInfo	user			= 2; //用户个人数据
	required BuildQueue			build_queue		= 3; //当前建造队列	
	repeated BuildInfo 			builds 			= 4; //建筑[数组]	
}

//免费加速
message BuildFreeSpeedReq
{
	required uint32 	id    			= 1;
}

message BuildFreeSpeedResp
{
	repeated BuildInfo 	build 			= 1;  //
	required BuildQueue queue		 	= 2;
}

//秒升
message BuildOverOnceReq
{
}

message BuildOverOnceResp
{
	repeated BuildInfo	builds			= 1;
	required uint32		hammer_cnt		= 2; //剩余锤子数量
	required BuildQueue	build_queue		= 3; //当前建造队列
	required uint32		cash			= 4; //钻石	
	required uint32		total_exploit	= 5; //累计功勋
}

//领取功勋
message BuildExploitReq
{
	
}

message BuildExploitResp
{
	required uint32 user_exploit			= 1; //玩家当前功勋
}

//自动升级
message BuildAutoUpgradeReq
{
	required uint32		type			= 1; //自动升级建筑类型
	required bool		handle			= 2; //[true--自动升级 false--取消升级]
}

message BuildAutoUpgradeResp
{	
	required BuildQueue	build_queue		= 1; //当前建造队列	
	repeated BuildInfo	builds			= 2;
	required BuildUserSyncInfo user	  	= 3;
}

//资源同步
message BuildResourceSyncReq
{
}

message BuildResourceSyncResq
{
	required uint32 coin 		= 1;//铜币
	required uint32 wood		= 2;//木材
	required uint32 food		= 3;//粮食
	required uint32 iron		= 4;//镔铁
}



